msktname.cpp

```diff
@@ -182,8 +182,18 @@ bool DnsSrvHost::validate(bool nocanon, std::string service) {
         }
         if (connect(sock, (struct sockaddr *) ai->ai_addr, ai->ai_addrlen) == -1) {
             int err = errno;
+            void *addr;
+            /* We assume max possible size */
             char addrstr[INET6_ADDRSTRLEN] = "";
-            (void) inet_ntop(ai->ai_family, ai->ai_addr, addrstr, INET6_ADDRSTRLEN);
+            switch (ai->ai_family) {
+                case AF_INET:
+                    addr = (void *) &(((struct sockaddr_in *) ai->ai_addr)->sin_addr);
+                    break;
+               case AF_INET6:
+                    addr = (void *) &(((struct sockaddr_in6 *) ai->ai_addr)->sin6_addr);
+                    break;
+            }
+            (void) inet_ntop(ai->ai_family, addr, addrstr, sizeof(addrstr));
```

Comment:
If we want to go agnostic to address families, how about something along the lines of:

```suggestion
            (void) getnameinfo(ai->ai_addr, ai->ai_addrlen, addrstr, sizeof(addrstr), NULL, 0, NI_NUMERICHOST | NI_NUMERICSERV);
```

and drop the addition of `addr` above. Bonus points for better error checking.

Reply:
Isn't `NI_NUMERICSERV` redundant since we aren't probing for a service here?

Reply:
Wait: You want to call this function twice, right?

Reply:
Correct. `getnameinfo()` when called with `NI_NUMERICHOST` is essentially just an AF-aware wrapper around `inet_ntop()`. It won't query DNS in this case. I've added `NI_NUMERICSERV` just to be sure, but having checked the [glibc implementation](https://github.com/lattera/glibc/blob/master/inet/getnameinfo.c) now, it's indeed redundant here.

Reply:
I can cofirm the same behavior in FreeBSD's libc impl.